cdap-docs/cdap-apps/source/hydrator/custom.rst

```diff
@@ -449,3 +449,255 @@ Plugin Deployment
+
+  Based on the above definitions we could write the configuration JSON for Stream source which has the following properties - schema, duration, name, format, delay - as
+
+  .. code-block:: Javascript
+
+    {
+      "metadata": {
+        "spec-version": "1.0"
+      },
+      "configuration-groups": [
+        {
+          "label": "Stream Configuration",
+          "properties": [
+            {
+              "widget-type": "textbox",
+              "label": "Stream Name",
+              "name": "name"
+            },
+            {
+              "widget-type": "textbox",
+              "label": "Duration",
+              "name": "duration"
+            },
+            {
+              "widget-type": "textbox",
+              "label": "Delay",
+              "name": "delay"
+            },
+            {
+              "widget-type": "textbox",
+              "label": "Data Field Name",
+              "name": "body.field",
+              "widget-attributes": {
+                "width": "medium"
+              }
+            },
+            {
+              "widget-type": "textbox",
+              "label": "Header Field Name",
+              "name": "headers.field",
+              "widget-attributes": {
+                "width": "medium"
+              }
+            },
+            {
+              "widget-type": "stream-selector",
+              "label": "Stream Name",
+              "name": "name",
+              "widget-attributes": {
+                "width": "medium"
+              }
+            }
+          ]
+        },
+        {
+          "label": "Format",
+          "properties": [
+            {
+              "widget-type": "select",
+              "label": "Format",
+              "name": "format",
+              "widget-attributes": {
+                "values": [
+                  "avro",
+                  "clf",
+                  "csv",
+                  "grok",
+                  "syslog",
+                  "text",
+                  "tsv"
+                ],
+                "default": "text"
+              }
+            }
+          ]
+        }
+      ],
+      "outputs": [
+        {
+          "name": "schema",
+          "widget-type": "schema",
+          "widget-attributes": {
+            "schema-types": [
+              "boolean",
+              "int",
+              "long",
+              "float",
+              "double",
+              "string",
+              "map<string, string>"
+            ],
+            "schema-default-type": "string",
+            "property-watch": "format"
+          }
+        }
+      ]
+    }
+
+FAQs
```

Comment:
Let's remove the FAQs. It is very contextual. The content is good. The questions needs to be more well defined. remove it for now. open a JIRA to address it in the next release.